Nama: Azis Muslim
Program: FTDS Batch 013 (Remote)

In [114]:
berat = 100
tinggi = 179
usia = 21

def kalkulator_BMR_L(berat, tinggi):
    BMR = 66 + (13.7*berat) + (5*tinggi) - (6.8*usia)
    if BMR > 2000:
        print("ideal")
    else:
        print("tidak ideal")
    


In [115]:
 BMR = 66 + (13.7*berat) + (5*tinggi) - (6.8*usia)

In [116]:
BMR

2188.2

In [118]:
import numpy as np
import pandas as pd

In [119]:
DataBMR = pd.read_csv("BMR.csv")

In [120]:
DataBMR

Age  Gender  Height (cm)  Weight (kg)  Index
0     21    Male          162           58      2
1     44  Female          188           80      2
2     45    Male          177           61      2
3     50    Male          192          101      3
4     28  Female          195           61      1
..   ...     ...          ...          ...    ...
145   33    Male          142           71      4
146   32  Female          162           58      2
147   29    Male          174           54      1
148   33  Female          197           72      2
149   23    Male          198           50      0

[150 rows x 5 columns]

In [121]:
DataBMR.loc[1:8].describe()

Age  Height (cm)  Weight (kg)     Index
count   8.000000     8.000000     8.000000  8.000000
mean   35.625000   180.750000    89.375000  2.750000
std    12.223368    13.739931    28.289006  1.488048
min    20.000000   152.000000    58.000000  1.000000
25%    25.000000   176.250000    61.000000  1.750000
50%    36.000000   184.000000    90.500000  2.500000
75%    45.750000   189.000000   109.000000  4.000000
max    50.000000   195.000000   132.000000  5.000000

In [122]:
DataBMR.groupby(by='Age')[['Index']].mean().sort_values(by='Index', ascending=False).head(5)

Index
Age       
25    3.25
26    3.25
47    3.00
29    3.00
38    2.80

Berdasarkan tabel diatas jika diurutkan dari BMR tertinggi ke yang terendah dapat diketahui secara umum usia 25 tahun memiliki index BMR tertinggi. dan usia yang paling tua disini 47 tahun index BMR nya berada di tengah. dari sini belum dapat disimpulkan apakah semakin tua kebutuhan energi dasarnya semakin tinggi atau tidak.

In [123]:
DataBMR.groupby(by='Age')[['Index']].mean().sort_values(by='Index', ascending=True).head(5)

Index
Age          
32   1.000000
21   1.333333
40   1.555556
45   1.666667
23   1.833333

Berdasarkan tabel diatas yang di kelompokkan dari nilai BMR terendah ke tertinggi dapat dilihat usia 23 tahun memiliki BMR tertinggi dari ke keempat kelompok usia lainnya namun belum bisa dikatakan semakin tua usia BMR nya semakin rendah karena ada kelompok umur 21 tahun yang nilai BMR nya kedua terendah dari lima kelompok usia.

In [124]:
DataBMR[
    (DataBMR["Gender"] == "Male") &
    (DataBMR["Index"] > 2.7 ) &
    (DataBMR["Age"] > 30)
]

Age Gender  Height (cm)  Weight (kg)  Index
3     50   Male          192          101      3
8     48   Male          174          107      4
28    40   Male          190           95      3
32    36   Male          156          137      5
34    34   Male          188          100      3
51    43   Male          194          115      4
74    44   Male          141           86      5
83    39   Male          188          123      4
101   37   Male          172          105      4
104   48   Male          166           70      3
109   31   Male          154          112      5
114   41   Male          161          115      5
121   42   Male          148           60      3
141   38   Male          199          156      4
142   35   Male          154          111      5
145   33   Male          142           71      4

In [125]:
DataBMR[
    (DataBMR["Gender"] == "Male") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].shape

(16, 5)

hasil diatas menjelaskan ada 16 baris untuk dataset Perempuan dan lima kolom dengan index BMR diatas 2.7 dan usia diatas 30

In [127]:
DataBMR[
    (DataBMR["Gender"] == "Male") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].head()

Age Gender  Height (cm)  Weight (kg)  Index
3    50   Male          192          101      3
8    48   Male          174          107      4
28   40   Male          190           95      3
32   36   Male          156          137      5
34   34   Male          188          100      3

In [128]:
Data_P = DataBMR[
    (DataBMR["Gender"] == "Male") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].head()

In [129]:
Data_P

Age Gender  Height (cm)  Weight (kg)  Index
3    50   Male          192          101      3
8    48   Male          174          107      4
28   40   Male          190           95      3
32   36   Male          156          137      5
34   34   Male          188          100      3

In [131]:
Data_P

Age Gender  Height (cm)  Weight (kg)  Index
3    50   Male          192          101      3
8    48   Male          174          107      4
28   40   Male          190           95      3
32   36   Male          156          137      5
34   34   Male          188          100      3

In [132]:
Data_P['BMR'] = 66 + (Data_P['Weight (kg)']*13.7) + (Data_P['Height (cm)']*5) - (Data_P['Age']*6.8)
Data_P.head()

Age Gender  Height (cm)  Weight (kg)  Index     BMR
3    50   Male          192          101      3  2069.7
8    48   Male          174          107      4  2075.5
28   40   Male          190           95      3  2045.5
32   36   Male          156          137      5  2478.1
34   34   Male          188          100      3  2144.8

In [133]:
Data_P.groupby(by='Age')[['BMR']].mean().sort_values(by='BMR', ascending=False).head(5)

BMR
Age        
36   2478.1
34   2144.8
48   2075.5
50   2069.7
40   2045.5

Berdasarkan tabel diatas dapat dilihat kelompuk Usia 36 Tahun memiliki Nilai BMR tertinggi diikuti dengan kelompok usia 34 tahun

In [113]:
Data_P.groupby(by='Age')[['BMR']].mean().sort_values(by='BMR', ascending=True).head(5)

BMR
Age        
40   2045.5
50   2069.7
48   2075.5
34   2144.8
36   2478.1

 Berdasarkan tabel iatas dapat dilihat bahwa kelompok usia 40 tahun memiliki BMR terendah di ikuti dengan kelompok usia 50 tahun

In [134]:
DataBMR[
    (DataBMR["Gender"] == "Female") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
]

Age  Gender  Height (cm)  Weight (kg)  Index
12    35  Female          190          118      4
16    33  Female          197          154      4
23    38  Female          186          127      4
77    33  Female          145           79      4
102   44  Female          156          106      5
103   47  Female          172           90      4
106   46  Female          177           81      3

In [135]:
DataBMR[
    (DataBMR["Gender"] == "Female") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].shape

(7, 5)

In [136]:
DataBMR[
    (DataBMR["Gender"] == "Female") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].head()

Age  Gender  Height (cm)  Weight (kg)  Index
12    35  Female          190          118      4
16    33  Female          197          154      4
23    38  Female          186          127      4
77    33  Female          145           79      4
102   44  Female          156          106      5

In [137]:
Data_W = DataBMR[
    (DataBMR["Gender"] == "Female") &
    (DataBMR["Index"] > 2.7) &
    (DataBMR["Age"] > 30)
].head()

In [138]:
Data_W

Age  Gender  Height (cm)  Weight (kg)  Index
12    35  Female          190          118      4
16    33  Female          197          154      4
23    38  Female          186          127      4
77    33  Female          145           79      4
102   44  Female          156          106      5

In [139]:
Data_W['BMR'] = 655 + (Data_W['Weight (kg)']*9.6) + (Data_W['Height (cm)']*1.8) - (Data_W['Age']*4.7)
Data_W.head()

Age  Gender  Height (cm)  Weight (kg)  Index     BMR
12    35  Female          190          118      4  1965.3
16    33  Female          197          154      4  2332.9
23    38  Female          186          127      4  2030.4
77    33  Female          145           79      4  1519.3
102   44  Female          156          106      5  1746.6

In [141]:
Data_W.groupby(by='Age')[['BMR']].mean().sort_values(by='BMR', ascending=False).head(5)

BMR
Age        
38   2030.4
35   1965.3
33   1926.1
44   1746.6

Bersarkan tabel diatas dapat dilihat kelompok Usia 38 tahun untuk perempuan memiliki BMR tertinggi di angka 2030.4 dan diikuti dengan kelompok usia 35 Tahun

In [142]:
Data_W.groupby(by='Age')[['BMR']].mean().sort_values(by='BMR', ascending=True).head(5)

BMR
Age        
44   1746.6
33   1926.1
35   1965.3
38   2030.4

Berdasarkan tabel diatas dapat dilihat kelompok usia 44 tahun untuk perempuan memiliki BMR terendah di angka 1746.6

#### Penjelasan Kelompok Klasifikasi BMR Perempuan dan BMR Laki-Laki

Untuk menyederhanakan analasi data. Data set dikelompokkan berdasarkan Gender yaitu Dataset untuk Laki-Laki yang dilambangkan dengan kode Data_P(Data Pria) dan Data_W(Data Wanita). 

Setelah dilakukan perhitungan BMR ditembukan bahwa BMR tertinggi untuk Laki-Laki dari data set ada di Usia 36 tahun dengan nilai BMR 2478 dan nilai terendahnya ada di usia 40 tahun dengan nilai BMR 2045.5

Pada pegelompokkan BMR bersadarkan Dataset perempuan ditemukan BMR tertinggi ada pada kelompok usia 38 tahun dengan nilai BMR 2030,4 dan nilai BMR terendahnya ada pada kelompok usia 44 tahun dengan nilai BMR 1746.6